packages/node/src/subcommands/reindex.service.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { Inject, Injectable } from '@nestjs/common';
+import {
+  getLogger,
+  MetadataFactory,
+  MetadataRepo,
+  MmrService,
+  NodeConfig,
+  StoreService,
+  SubqueryRepo,
+} from '@subql/node-core';
+import { Sequelize } from 'sequelize';
+import { SubqueryProject } from '../configure/SubqueryProject';
+import {
+  getExistingProjectSchema,
+  getMetaDataInfo,
+  initDbSchemaUtil,
+} from '../utils/project';
+
+const logger = getLogger('Reindex');
+
+@Injectable()
+export class ReindexService {
+  private _schema: string;
+  private metadataRepo: MetadataRepo;
+
+  constructor(
+    private readonly sequelize: Sequelize,
+    private readonly nodeConfig: NodeConfig,
+    private readonly storeService: StoreService,
+    private readonly mmrService: MmrService,
+    private readonly project: SubqueryProject,
+    @Inject('Subquery') protected subqueryRepo: SubqueryRepo,
+  ) {}
+
+  private async getExistingProjectSchema(): Promise<string> {
+    return getExistingProjectSchema(
+      this.nodeConfig,
+      this.sequelize,
+      this.subqueryRepo,
+    );
+  }
+
+  get schema(): string {
+    return this._schema;
+  }
+
+  private async getLastProcessedHeight(): Promise<number | undefined> {
+    return getMetaDataInfo(this.metadataRepo, 'lastProcessedHeight');
+  }
+
+  private async getMetadataBlockOffset(): Promise<number | undefined> {
+    return getMetaDataInfo(this.metadataRepo, 'blockOffset');
+  }
+
+  private async initDbSchema(): Promise<void> {
+    await initDbSchemaUtil(this.project, this.schema, this.storeService);
+  }
+
+  async reindex(targetBlockHeight: number): Promise<void> {
```

Comment:
Any chance we can do this in here too? https://github.com/subquery/subql/issues/1230